### ISSUE TITLE
99-mellanox.sh script broken when there is no "/sys/class/infiniband_cm"

### ISSUE THREAD

**Issue by jasonguy** [NVIDIA/enroot#118]
The script [99-mellanox.sh ](https://github.com/NVIDIA/enroot/blob/master/conf/hooks/99-mellanox.sh) breaks on hosts with newer linux-rdma package. It looks like [this commit](https://patchwork.kernel.org/project/linux-rdma/patch/0-v1-90096a98c476+205-remove_cm_leftovers_jgg@nvidia.com/) in the linux kernel removed all references to the /sys/class/infiniband_cm/
directory represented by the cm_class symbol.

**Comment by flx42** [comment]
Which version of enroot are you using? This should have been fixed by https://github.com/NVIDIA/enroot/commit/d80b3f6749bd796d05c136bb30788e7acd63afd6

**Comment by jasonguy** [comment]
Looks like the user (not me) was following “Running with Pyxis/Enroot” in [NVIDIA HPC-Benchmarks page](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/hpc-benchmarks). Command executed on the DGX was:
`root@p-dgx-a100-009:/home/secure# srun -N 1 --ntasks-per-node=8 --cpu-bind=none --mpi=none --container-image="/home/secure/hpl-baai-21.4.sqsh" hpl.sh --xhpl-ai --config dgx-a100 --dat /workspace/hpl-ai-linux-x86_64/sample-dat/HPL-dgx-a100-1N.dat`

**Comment by flx42** [comment]
Right, but this doesn't tell me which enroot version it is. You need to run:
```console
$ enroot version
3.3.0
```

**Comment by jasonguy** [comment]
Looks like the answer is in the error log...

slurmstepd: error: pyxis: enroot-mount: failed to mount: /sys/class/infiniband_cm/abi_version at /root/.local/share/**enroot/pyxis_19.0**/sys/class/infiniband_cm/abi_version: No such file or directory
slurmstepd: error: pyxis: [ERROR] /usr/local/etc/enroot/hooks.d/99-mellanox.sh exited with return code 1

**Comment by flx42** [comment]
No, this is just the Slurm job ID.
